lib/puppet/type/elasticsearch_snapshot_repository.rb

```diff
@@ -0,0 +1,52 @@
+$LOAD_PATH.unshift(File.join(File.dirname(__FILE__), '..', '..'))
+
+require 'puppet_x/elastic/elasticsearch_rest_resource'
+
+# rubocop:disable Metrics/BlockLength
+Puppet::Type.newtype(:elasticsearch_snapshot_repository) do
+  extend ElasticsearchRESTResource
+
+  desc 'Manages Elasticsearch snapshot repositories.'
+
+  ensurable
+
+  newparam(:name, :namevar => true) do
+    desc 'Repository name.'
+  end
+
+  newparam(:type) do
+    desc 'Repository type'
+    defaultto 'fs'
+
+    validate do |value|
+      raise Puppet::Error, 'string expected' unless value.is_a? String
+    end
+  end
+
+  # newproperty(:compress, :boolean => true, :parent => Puppet::Property::Boolean) do
+  newproperty(:compress, :boolean => true) do
+    desc 'Compress the repository data'
+
+    defaultto :true
+  end
+
+  newproperty(:location) do
+    desc 'Repository location'
+  end
+
+  newproperty(:chunk_size) do
+    desc 'File chunk size'
+  end
+
+  newproperty(:max_restore_rate) do
+    desc 'Maximum Restore rate'
+  end
+
+  newproperty(:max_snapshot_rate) do
+    desc 'Maximum Snapshot rate'
+  end
+
+  # rubocop:disable Style/SignalException
+  validate do
```

Comment:
Was this leftover?

Reply:
Yeh, currently there's no validation... 

I'm wondering if there should be some though... Thoughts?

Reply:
Maybe it could verify that the snapshot repository has a `location` set, since that's always required AFAIK.